src/sinol_make/commands/run/__init__.py

```diff
@@ -397,135 +575,51 @@ def run_solutions(self, compiled_commands, names, solutions, report_file):
-                # if self.args.verbose:
-                #   print_stream(6*" ", end=" | ")
-                #   for program in program_group:
-                #       print_stream(10*" ", end=" | ")
-                #   print_stream()
-                #   for test in self.tests:
-                #       print_stream("%6s" % self.extract_test_id(test), end=" | ")
-                #       for program in program_group:
-                #           result = all_results[program][self.get_group(test)][test]
-                #           status = result.Status
-                #           if status == "  ": print_stream(10*' ', end=" | ")
-                #           else:
-                #               print_stream("%3s" % self.colorize_status(status),
-                #                   ("%17s" % self.color_time(result.Time, self.time_limit)) if getattr(result, "Time") is not None else 7*" ", end=" | ")
-                #       print_stream()
-                #       if not self.args.hide_memory:
-                #           print_stream(6*" ", end=" | ")
-                #           for program in program_group:
-                #               result = all_results[program][self.get_group(test)][test]
-                #               print_stream(("%20s" % self.color_memory(result.Memory, self.memory_limit))  if getattr(result, "Memory") is not None else 10*" ", end=" | ")
-                #           print_stream()
-                #   print_stream()
-                print_stream(10*len(program_group)*' ')
-
-            if output_file is not None:
-                os.system('sed -i -r "s/\x1B\[([0-9]{1,3}(;[0-9]{1,2})?)?[mGK]//g" %s' % output_file) # TODO: make this work on Windows
-                print("Report has been saved to", util.bold(output_file))
-                print()
-
-        print("Performing %d executions..." % len(executions))
-        with mp.Pool(self.cpus) as pool:
+        has_terminal = True
+        try:
+            terminal_width = os.get_terminal_size().columns
+            terminal_height = os.get_terminal_size().lines
+        except OSError:
+            has_terminal = False
+            terminal_width = 80
+            terminal_height = 30
+
+        if has_terminal:
+            run_event = threading.Event()
+            run_event.set()
+            thr = threading.Thread(target=printer.printer_thread, args=(run_event, print_view, program_groups_scores,
+                                                                        all_results, print_data, names, executions,
+                                                                        self.groups, self.scores, self.tests,
+                                                                        self.possible_score, self.time_limit,
+                                                                        self.memory_limit, self.cpus,
+                                                                        self.args.hide_memory))
```

Comment:
That's not the ideal intendation, I guess.

Reply:
I changed the intendation.

Reply:
I haven't checked, but the lines still seem to be over 120 characters long. Do they fit, or are they over 120?

Reply:
They're under 120 lines

Reply:
characters* not lines :sweat_smile:

Reply:
Yes, that's what I meant